vc-worker.py

```diff
@@ -161,59 +232,27 @@ def action_others(conn, action, vc):
         elif request_type == 'stop':
             action_result = action_stop(request, machine_ro, vc)
         elif request_type == 'get_info':
-            logger.debug(request.to_dict())
-            try:
-                info = vc.get_machine_info(machine_ro.provider_id)
-                logger.debug(info)
-            except Exception:
-                logger.error('get_info exception: ', exc_info=True)
-                info = {'ip_addresses': [], 'nos_id': '', 'machine_search_link': ''}
-
-            machine = data.Machine.get_one_for_update({'_id': request.machine}, conn=conn)
-            machine.nos_id = info['nos_id']
-            machine.machine_search_link = info['machine_search_link']
-
-            if len(info['ip_addresses']) != 0:
-                machine.ip_addresses = info['ip_addresses']
-                machine.save(conn=conn)
-                request.state = 'success'
-                action.lock = -1
-            else:
-                machine.save(conn=conn)
-                request.state = 'delayed'
-                action.repetitions -= 1
-                action.next_try = datetime.datetime.now() + datetime.timedelta(
-                    seconds=random.randint(action.delay, action.delay+3)
-                )
-                action.lock = 1
-            request.save(conn=conn)
-            action.save(conn=conn)
+            action_get_info(request, machine_ro, vc, action, conn)
             return
+        elif request_type == 'take_screenshot':
+            action_result = action_take_screenshot(request, machine_ro, vc, action, conn)
         else:
             logger.warn('unknown request: {} is going to succeed'.format(request_type))
-        machine = data.Machine.get_one_for_update({'_id': request.machine}, conn=conn)
-        machine.state = action_result['machine.state']
+
+        if request_type in ['undeploy', 'start', 'stop']:
+            machine = data.Machine.get_one_for_update({'_id': request.machine}, conn=conn)
+            if machine.state not in ['undeployed', 'failed']:
```

Comment:
I would like to kindly ask for adding comment, why these two states are excluded from change of state